### ISSUE TITLE
Unclear needed inputs in the config.json file

### ISSUE THREAD

**Issue by dalonsoa** [alan-turing-institute/binderhub-deploy#12]
The required inputs in the `config.json` file are not entirely clear, and might even be misleading:

- Should `node_count` be a number? Not discussed in the workshop. 
- `vm_type` should be a complete vm name like Standard_A8m_v2,Standard_A9, etc. Where these EXACT names can be found? The linked page only indicates "commercial" names.
- `location` should be `region`, as locations are cities and not the input `az` expects. And it's unclear what to do with the spaces: West Europe --> westeurope. 
- What binderhub.version should be? Is an arbitrary version number chosen by you or the actual helm version number you want to use? This is not chosen in the Workshop instructions.
- Maybe that section should be renamed `helm` instead of `binderhub`.

**Comment by sgibson91** [comment]
> * Should `node_count` be a number? Not discussed in the workshop.

Node count is passed to `az aks create` in [step 5](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md#5-create-an-azure-container-service-aks-cluster).

**Comment by dalonsoa** [comment]
Ups! I missed that one.

**Comment by sgibson91** [comment]
No worries, I'll still update the README so it's clearer 🙂

**Comment by sgibson91** [comment]
> * `vm_type` should be a complete vm name like Standard_A8m_v2,Standard_A9, etc. Where these EXACT names can be found? The linked page only indicates "commercial" names.

I don't know the answer to this. Might be my own ignorance or maybe a gap in the documentation. @trallard , @leestott ?

**Comment by sgibson91** [comment]
> * `location` should be `region`, as locations are cities and not the input `az` expects. And it's unclear what to do with the spaces: West Europe --> westeurope.

@dalonsoa so we have a choice to make here. While I agree that the Azure webpage says these are regions, this value is parsed to `az group create` via the `--location` flag (see [step](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md#3-create-a-resource-group)). Hence it makes sense for parsing to have `--location ${location}` rather than `--location ${region}`.

@tmbgreaves and I will check the new parser to make sure that West Europe and westeurope are parsed identically, but you would deal with that as so:
```
az group create --name resource-group --location "West Europe" --output table
```
```
az group create --name resource-group --location westeurope --output table
```

**Comment by dalonsoa** [comment]
I agree, but I think that the point is that if we say "location" people will check the web and use "London" for example, which is a location. And when running the command, that will produce an error: 

`The provided location 'London' is not available for resource group. List of available regions is 'centralus,eastasia,southeastasia,eastus,eastus2,westus,westus2,northcentralus,southcentralus,westcentralus,northeurope,westeurope,japaneast,japanwest,brazilsouth,australiasoutheast,australiaeast,westindia,southindia,centralindia,canadacentral,canadaeast,uksouth,ukwest,koreacentral,koreasouth,francecentral,southafricawest,southafricanorth'.`

**Comment by dalonsoa** [comment]
It might be enough to clarify in the documentation they should use the "Region" name, and not the location, but it seems a bit convoluted.

**Comment by sgibson91** [comment]
> * For a list of available data centre locations, [see here](https://azure.microsoft.com/en-us/global-infrastructure/locations/).

Does this part of the README not mitigate that?

**Comment by sgibson91** [comment]
[`201e336`](https://github.com/alan-turing-institute/binderhub-deploy/commit/201e3366801af839842b4733391274c0dd0a913a) 👈 @dalonsoa what do you think?

**Comment by sgibson91** [comment]
> * What binderhub.version should be? Is an arbitrary version number chosen by you or the actual helm version number you want to use? This is not chosen in the Workshop instructions.

The link to page containing the chart versions was provided in [this step](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md#3-install-binderhub), and has now been added to the README in this project. (I think I realised at the same you did that it wasn't there 😞). I'll add a bit more description so it's clearer.

> * Maybe that section should be renamed `helm` instead of `binderhub`.

I have strong feelings against this as this key also contains the name that will be assigned to the BinderHub when it's deployed, but open to discussing it more ✨

**Comment by sgibson91** [comment]
[`0cafb99`](https://github.com/alan-turing-institute/binderhub-deploy/commit/0cafb99f5b5cc5c41d05dd9cedba86ceeecc0596) @dalonsoa good?

**Comment by sgibson91** [comment]
> > * `vm_type` should be a complete vm name like Standard_A8m_v2,Standard_A9, etc. Where these EXACT names can be found? The linked page only indicates "commercial" names.
> 
> I don't know the answer to this. Might be my own ignorance or maybe a gap in the documentation. @trallard , @leestott ?

This has been addressed in [`997aa90`](https://github.com/alan-turing-institute/binderhub-deploy/pull/11/commits/997aa9050ea081e34460a6336b25c35976c8dfcf)

**Comment by dalonsoa** [comment]
Almost complete! This is just for the General Purpose machines, and only for Windows VM. There's also memory optimised, storage optimised, etc. in addition to Linux. 

I would use the parent page (https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes?toc=%2fazure%2fvirtual-machines%2fwindows%2ftoc.json) and still put an example name, to clarify.
